sys/net/network_layer/fib/fib.c

```diff
@@ -363,10 +363,12 @@ static int fib_signal_rp(fib_table_t *table, uint16_t type, uint8_t *dat,
                 }
             }
             else {
-                fib_sr_t *temp_sr = (fib_sr_t *)dat;
-                size_t dat_size_in_bits = temp_sr->sr_dest->address->address_size << 3;
+                fib_sr_t temp_sr;
+                assert(dat_size >= sizeof(temp_sr));
+                memcpy(&temp_sr, dat, sizeof(temp_sr));
```

Comment:
Why copy the whole `fib_sr_t` if we only are interested in `sr_dest`?

```C
fib_sr_entry_t *sr_dest;
memcpy(sr_dest, &temp_sr->sr_dest, sizeof(sr_dest));
```

Reply:
Like this?